docs/multiregion.md

```diff
@@ -0,0 +1,98 @@
+
+For truly global scoped KubeCarrier installation current architecture isn't satisfactory. Here are some of the problems honest and hard-working system operator could encounter:
+
+* latency --> It's as simple as managing `eu-central` from a `east-asia` region would take between 200 and 400ms. This may or may not be satisfactory for the system requirement, but it's a matter of life. And speed of light. Certain operations require multiple round trips, increasing the overall latency.
+* Failure domains --> Failure of management cluster, or its unreachability would make the whole system immutable. However unlikely the whole management cluster becomes unavailable, it's a possible failure mode
+* Legalese & policy --> for some legal requirements/policy certain things need be separate.
+
+## Proposed solution -- introducing region concept
+
+To solve the above mentioned issues, we'll introduce managed kubecarrier federation. The basic concepts would be:
+
+* ServiceCluster --> stays the same as before; its akin to availability zone.
+* Region --> a group of multiple providers & service clusters regionally grouped with reasonable low-latency network connection is a region. The region has a single KubeCarrier management cluster
+
+This would provide the logical, geographical, and failure domain grouping to acchive the initial motivational goals. How to coordinate and configure the global KubeCarrier solution from single place (or to the end user appearing as a single place) is still an open problem.
+
+Between regions the following things should be synced, and should be globally unique:
+* `(ServiceCluster, Account)`
+
+Additionally the following objects should be replicated across every region:
+* `(Account)`
+
+Thus there must exist global consensus what the global KubeCarrier configuration should be. The authorization shall be handled the same as in the single region case --> management cluster shall create the necessary roles and rolebinding for the account.
+
+This is enough to handle the authorization via the native k8s mechanism. For Authentication see the dedicated section.
+
+## related work:
+
+* https://github.com/kubermatic/kubecarrier/pull/291/files
+
+This solutions revolves around having a single superCluster managing multiple regions with, and each region being managed by region-local installation.
+
+Pros:
+* the user authentication happens at the superCluster level, whilst their actions is propagated to region-local management cluster via the `Impersonate` headers
+* installation is handled via the superCluster operator
+
+Cons:
+* the latency issue still remains. (e.g. request for creating new service instance would propagate from user -- superClusterManager -- region -- region local serviceCluster)
+* the superCluster becomes Single point of Failure
```

Comment:
Can we handle that by increasing the replicas of the supercluster installations?

Reply:
I'm talking about region failure, not a single node failure. It's unlikely, but not an impossible scenario.